### PR TITLE
Introduce cdk to repository

### PR DESCRIPTION
**Pull request by JuliaBrigitte**
Co-authored-by: Nic <nicolas.long@guardian.co.uk>

## What does this change?

Introduce cdk to repository preparing for migration

## How to test

```
npm test 
npm run difflocal
```

### CODE REVIEW

**Comment by nicl** [comment]
@ioannakok @bryophyta , @jamesgorrie mentioned you might be good people to review this. We've begun migrating this lambda to CDK to help onboard @JuliaBrigitte and also provide an example to Dotcom here. Really happy to talk through things if helpful so do shout.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        with:
+        role-to-assume: ${{ secrets.GU_RIFF_RAFF_ROLE_ARN }}
+        aws-region: eu-west-1
+      - uses: actions/setup-java@v3
+        with:
+          java-version: "11"
+          distribution: "corretto"
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
```
Comment (akash1810) [inline .github/workflows/ci.yaml:26]: Could we source the node version from a `.nvmrc` file? Else we'd have to remember to keep the two in-sync.

**Review by akash1810 — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+  name; CI
```
Comment (AshCorr) [inline .github/workflows/ci.yaml:1]:
```suggestion
  name: CI
```

Is this a typo?

**Review by AshCorr — CHANGES_REQUESTED** [review]

File: cdk/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    "noUnusedParameters": false,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": false,
+    "inlineSourceMap": true,
+    "inlineSources": true,
+    "experimentalDecorators": true,
+    "strictPropertyInitialization": false,
+    "typeRoots": ["./node_modules/@types"],
+    "outDir": "dist"
+  },
```
Comment (mxdvl) [inline cdk/tsconfig.json:29]: Could we extend from `@guardian/tsconfig`?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (mxdvl) [inline .github/workflows/ci.yaml:26]: Could this come from an `.nvmrc` node version file?

```suggestion
      - uses: actions/setup-node@v3
        with:
          node-version-file: ".nvmrc"
```

File: cdk/lib/facia-purger.test.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+describe('The FaciaPurger stack', () => {
+	it('matches the snapshot', () => {
+		const app = new App();
+		const stack = new FaciaPurger(app, 'FaciaPurger', {
+			stack: 'frontend',
+			stage: 'TEST',
+		});
+		const template = Template.fromStack(stack);
+		expect(template.toJSON()).toMatchSnapshot();
```
Comment (mxdvl) [inline cdk/lib/facia-purger.test.ts:13]: How do you regenerate a snapshot?

**Review by mxdvl — APPROVED** [review]
I get it now!

This adds the CDK to the repository and creates an [initial snapshot](./cdk/lib/__snapshots__/facia-purger.test.ts.snap) of the current `cloudformation.yaml`.

In #25 we then replace the snapshot with a new cloudformation generated from guCDK.

---

- [AWS docs on snapshot testing](https://docs.aws.amazon.com/cdk/v2/guide/testing.html#testing_snapshot)

File: cdk/lib/facia-purger.test.ts (hunk 4 above)
Comment (akash1810) [reply cdk/lib/facia-purger.test.ts:13]: https://jestjs.io/docs/snapshot-testing#updating-snapshots

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (mxdvl) [reply .github/workflows/ci.yaml:26]: Non-blocking: update this to an `.nvmrc` node version file!

```suggestion
      - uses: actions/setup-node@v3
        with:
          node-version-file: ".nvmrc"
```

**Review by AshCorr — APPROVED** [review]
Woop!

**Pull request merged**